lib/parseSty.js

```diff
@@ -0,0 +1,101 @@
+"use strict";
+
+var starts = ["\\newcommand",
+              "\\renewcommand",
+              "\\DeclareRobustCommand",
+              "\\defSpecFun"];
+
+function findBracketSections(string) {
+    return string.match(/\{[^\{\}]*?(([^\}]*)\}+)*?[^\{\}]*?\}|\[[^\[\]]*?(([^\]]*)\]+)*?[^\[\]]*?\]/g).map(function(s) {
```

Comment:
this regexp is hard to read, and would need at least a comment. Also take into account my earlier comment about using a library.